<a href="https://colab.research.google.com/github/Deepikagh97/Telecom-Churn-Analysis/blob/main/DN_%7C_Telecom_Churn_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Telecom Churn Analysis**

In [2]:
# import all librariers
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
sns.set(color_codes=True)

import warnings
warnings.filterwarnings("ignore")

### **Read and view data**

In [3]:
# Read the dataset
from google.colab import files
uploaded = files.upload()
df = pd.read_csv("Telecom Churn.csv")

Saving Telecom Churn.csv to Telecom Churn.csv


In [17]:
#Glimpse of Data
df.head()

State  Account length  Area code International plan Voice mail plan  \
0    KS             128        415                 No             Yes   
1    OH             107        415                 No             Yes   
2    NJ             137        415                 No              No   
3    OH              84        408                Yes              No   
4    OK              75        415                Yes              No   

   Number vmail messages  Total day minutes  Total day calls  \
0                     25              265.1              110   
1                     26              161.6              123   
2                      0              243.4              114   
3                      0              299.4               71   
4                      0              166.7              113   

   Total day charge  Total eve minutes  Total eve calls  Total eve charge  \
0             45.07              197.4               99             16.78   
1             27.47              195.5              103             16.62   
2             41.38              121.2              110             10.30   
3             50.90               61.9               88              5.26   
4             28.34              148.3              122             12.61   

   Total night minutes  Total night calls  Total night charge  \
0                244.7                 91               11.01   
1                254.4                103               11.45   
2                162.6                104                7.32   
3                196.9                 89                8.86   
4                186.9                121                8.41   

   Total intl minutes  Total intl calls  Total intl charge  \
0                10.0                 3               2.70   
1                13.7                 3               3.70   
2                12.2                 5               3.29   
3                 6.6                 7               1.78   
4                10.1                 3               2.73   

   Customer service calls  Churn  
0                       1  False  
1                       1  False  
2                       0  False  
3                       2  False  
4                       3  False

In [18]:
df.tail()

State  Account length  Area code International plan Voice mail plan  \
3328    AZ             192        415                 No             Yes   
3329    WV              68        415                 No              No   
3330    RI              28        510                 No              No   
3331    CT             184        510                Yes              No   
3332    TN              74        415                 No             Yes   

      Number vmail messages  Total day minutes  Total day calls  \
3328                     36              156.2               77   
3329                      0              231.1               57   
3330                      0              180.8              109   
3331                      0              213.8              105   
3332                     25              234.4              113   

      Total day charge  Total eve minutes  Total eve calls  Total eve charge  \
3328             26.55              215.5              126             18.32   
3329             39.29              153.4               55             13.04   
3330             30.74              288.8               58             24.55   
3331             36.35              159.6               84             13.57   
3332             39.85              265.9               82             22.60   

      Total night minutes  Total night calls  Total night charge  \
3328                279.1                 83               12.56   
3329                191.3                123                8.61   
3330                191.9                 91                8.64   
3331                139.2                137                6.26   
3332                241.4                 77               10.86   

      Total intl minutes  Total intl calls  Total intl charge  \
3328                 9.9                 6               2.67   
3329                 9.6                 4               2.59   
3330                14.1                 6               3.81   
3331                 5.0                10               1.35   
3332                13.7                 4               3.70   

      Customer service calls  Churn  
3328                       2  False  
3329                       3  False  
3330                       2  False  
3331                       2  False  
3332                       0  False

In [19]:
print('The number of rows are',df.shape[0],'\n''The number of columns are',df.shape[1])

The number of rows are 3333 
The number of columns are 20


***Checking datatype of all***

In [20]:
df.dtypes

State                      object
Account length              int64
Area code                   int64
International plan         object
Voice mail plan            object
Number vmail messages       int64
Total day minutes         float64
Total day calls             int64
Total day charge          float64
Total eve minutes         float64
Total eve calls             int64
Total eve charge          float64
Total night minutes       float64
Total night calls           int64
Total night charge        float64
Total intl minutes        float64
Total intl calls            int64
Total intl charge         float64
Customer service calls      int64
Churn                        bool
dtype: object

There are 20 columns out of one column "Churn" which is the target variable is of Boolean type. There are 3 object type and 16 of numerical type.

***Checking information about the dataset***

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3333 entries, 0 to 3332
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   State                   3333 non-null   object 
 1   Account length          3333 non-null   int64  
 2   Area code               3333 non-null   int64  
 3   International plan      3333 non-null   object 
 4   Voice mail plan         3333 non-null   object 
 5   Number vmail messages   3333 non-null   int64  
 6   Total day minutes       3333 non-null   float64
 7   Total day calls         3333 non-null   int64  
 8   Total day charge        3333 non-null   float64
 9   Total eve minutes       3333 non-null   float64
 10  Total eve calls         3333 non-null   int64  
 11  Total eve charge        3333 non-null   float64
 12  Total night minutes     3333 non-null   float64
 13  Total night calls       3333 non-null   int64  
 14  Total night charge      3333 non-null   

***Now, let us check the basic measures of descriptive statistics for the continuous variables***

In [22]:
df.describe(include = 'all').T

count unique    top  freq        mean        std  \
State                     3333     51     WV   106         NaN        NaN   
Account length          3333.0    NaN    NaN   NaN  101.064806  39.822106   
Area code               3333.0    NaN    NaN   NaN  437.182418   42.37129   
International plan        3333      2     No  3010         NaN        NaN   
Voice mail plan           3333      2     No  2411         NaN        NaN   
Number vmail messages   3333.0    NaN    NaN   NaN     8.09901  13.688365   
Total day minutes       3333.0    NaN    NaN   NaN  179.775098  54.467389   
Total day calls         3333.0    NaN    NaN   NaN  100.435644  20.069084   
Total day charge        3333.0    NaN    NaN   NaN   30.562307   9.259435   
Total eve minutes       3333.0    NaN    NaN   NaN  200.980348  50.713844   
Total eve calls         3333.0    NaN    NaN   NaN  100.114311  19.922625   
Total eve charge        3333.0    NaN    NaN   NaN    17.08354   4.310668   
Total night minutes     3333.0    NaN    NaN   NaN  200.872037  50.573847   
Total night calls       3333.0    NaN    NaN   NaN  100.107711  19.568609   
Total night charge      3333.0    NaN    NaN   NaN    9.039325   2.275873   
Total intl minutes      3333.0    NaN    NaN   NaN   10.237294    2.79184   
Total intl calls        3333.0    NaN    NaN   NaN    4.479448   2.461214   
Total intl charge       3333.0    NaN    NaN   NaN    2.764581   0.753773   
Customer service calls  3333.0    NaN    NaN   NaN    1.562856   1.315491   
Churn                     3333      2  False  2850         NaN        NaN   

                          min    25%    50%    75%    max  
State                     NaN    NaN    NaN    NaN    NaN  
Account length            1.0   74.0  101.0  127.0  243.0  
Area code               408.0  408.0  415.0  510.0  510.0  
International plan        NaN    NaN    NaN    NaN    NaN  
Voice mail plan           NaN    NaN    NaN    NaN    NaN  
Number vmail messages     0.0    0.0    0.0   20.0   51.0  
Total day minutes         0.0  143.7  179.4  216.4  350.8  
Total day calls           0.0   87.0  101.0  114.0  165.0  
Total day charge          0.0  24.43   30.5  36.79  59.64  
Total eve minutes         0.0  166.6  201.4  235.3  363.7  
Total eve calls           0.0   87.0  100.0  114.0  170.0  
Total eve charge          0.0  14.16  17.12   20.0  30.91  
Total night minutes      23.2  167.0  201.2  235.3  395.0  
Total night calls        33.0   87.0  100.0  113.0  175.0  
Total night charge       1.04   7.52   9.05  10.59  17.77  
Total intl minutes        0.0    8.5   10.3   12.1   20.0  
Total intl calls          0.0    3.0    4.0    6.0   20.0  
Total intl charge         0.0    2.3   2.78   3.27    5.4  
Customer service calls    0.0    1.0    1.0    2.0    9.0  
Churn                     NaN    NaN    NaN    NaN    NaN

The average 'Number vmail messages' is 8.099010, 'Total day minutes' is 179.77, 'Total day calls' is 100.435644, 'Total day charge' is 30.562307, 'Total eve minutes' is 200.980348, 'Total eve calls' is 100.114311, 'Total eve charge' is 17.083540, Total night minutes' is 200.872037, 'Total night calls' is 100.107711, 'Total night charge' is 9.039325, Total intl minutes' is 10.237294, 'Total intl calls' is 4.479448, 'Total intl charge' is 2.764581, 'Customer service calls' is 1.562856.

Avarage of Total day charge is more than total eve charge, total night charge, and total intl charge.

Around 50% of customers didn't used vmail messages.

There are 51 states.

In [23]:
print(df.isnull().sum())

State                     0
Account length            0
Area code                 0
International plan        0
Voice mail plan           0
Number vmail messages     0
Total day minutes         0
Total day calls           0
Total day charge          0
Total eve minutes         0
Total eve calls           0
Total eve charge          0
Total night minutes       0
Total night calls         0
Total night charge        0
Total intl minutes        0
Total intl calls          0
Total intl charge         0
Customer service calls    0
Churn                     0
dtype: int64


There are no null values present in the dataset.